packages/lib-utils/src/k8s/hooks/useK8sWatchResource.ts

```diff
@@ -23,12 +24,14 @@ const NOT_A_VALUE = '__not-a-value__';
  *   const watchRes = {
         ...
       }
- *   const [data, loaded, error] = UseK8sWatchResource(watchRes)
+ *   const [data, loaded, error] = useK8sWatchResource(watchRes)
  *   return ...
  * }
  * ```
  */
-export const useK8sWatchResource: UseK8sWatchResource = (initResource) => {
+export const useK8sWatchResource = <R extends K8sResourceCommon | K8sResourceCommon[]>(
+  initResource: WatchK8sResource | null,
```

Comment:
@vidyanambiar i'm not sure we want `| null` here. I think we'd want to enforce a K8s resource is always passed in - as it doesn't make much sense to watch nothing (unless I'm missing something). We can follow up on this in our TS cleanup: https://issues.redhat.com/browse/HAC-801

Reply:
That's a good point, Bryan. I think I introduced this in the commit that was attempting to resolve the issue with `any`  for the plural hook, but this isn't needed. I'll get rid of it.